src/drawing/DrawingCreator.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+import * as uuid from 'uuid';
+import DrawingPath, { DrawingPathRef, getPathCommands } from './DrawingPath';
+import DrawingSVG, { DrawingSVGRef } from './DrawingSVG';
+import PointerCapture, { Status as DrawingStatus } from '../components/PointerCapture';
+import { getStrokeWidths } from './DrawingPathGroup';
+import { PathGroup, Position, Stroke } from '../@types';
+import './DrawingCreator.scss';
+
+type Props = {
+    className?: string;
+    onAbort?: () => void;
+    onStart: () => void;
+    onStop: (pathGroup: PathGroup) => void;
+    stroke?: Stroke;
+};
+
+const defaultStroke = {
+    color: '#4826c2',
+    size: 4,
+};
+
+export default function DrawingCreator({
+    className,
+    onAbort = noop,
+    onStart,
+    onStop,
+    stroke = defaultStroke,
+}: Props): JSX.Element {
+    const [drawingStatus, setDrawingStatus] = React.useState<DrawingStatus>(DrawingStatus.init);
+    const capturedPathRef = React.useRef<Array<Position>>([]);
+    const creatorElRef = React.useRef<HTMLDivElement>(null);
+    const drawingDirtyRef = React.useRef<boolean>(false);
+    const drawingPathRef = React.useRef<DrawingPathRef>(null);
+    const drawingSVGRef = React.useRef<DrawingSVGRef>(null);
+    const renderHandleRef = React.useRef<number | null>(null);
+
+    const getPathGroup = (): PathGroup | null => {
+        const { current: creatorEl } = creatorElRef;
+        const { current: points } = capturedPathRef;
```

Comment:
ABC

Reply:
Seems like it already is?

Reply:
Oh I thought the right hand side of the assignment was supposed to be alphabetized (captured before creatorElRef).